clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -3789,15 +3789,26 @@ static void emitCheckHandlerCall(CodeGenFunction &CGF,
   bool NeedsAbortSuffix =
       IsFatal && RecoverKind != CheckRecoverableKind::Unrecoverable;
   bool MinimalRuntime = CGF.CGM.getCodeGenOpts().SanitizeMinimalRuntime;
+  // The preserve-all logic is somewhat duplicated in BoundsChecking.cpp for
+  // local-bounds. Make sure to change that too.
+  const auto &T = CGF.CGM.getTriple();
+  bool HandlerPreserveAllRegs =
+      CGF.CGM.getCodeGenOpts().SanitizeHandlerPreserveAllRegs &&
```

Comment:
Same, can you make sure SanitizeHandlerPreserveAllRegs already set only on supported arches?

Reply:
> Eg. there are some UBSAN/sanitizer file which flip on/of features deppending on context
your patch is exporting the logic outside that thing

Please clarify